pkg/cli/options_test.go

```diff
@@ -34,6 +35,86 @@ import (
 )
 
 var _ = Describe("Discover external plugins", func() {
+	Context("validate plugins root path", func() {
+		var (
+			homePath   string = os.Getenv("HOME")
+			customPath string = "/tmp/myplugins"
+			xdghome    string = os.Getenv("XDG_CONFIG_HOME")
+		)
+
+		Context("when using the $HOME environment variable", func() {
+			BeforeEach(func() {
+				err := os.Unsetenv("KB_PLUGINS_PATH")
+				Expect(err).To(BeNil())
+			})
+
+			It("should return the correct path for the darwin OS", func() {
+				plgPath, err := getPluginsRoot("darwin")
+				Expect(err).To(BeNil())
+				if xdghome != "" {
+					Expect(plgPath).To(Equal(fmt.Sprintf("%s/kubebuilder/plugins", xdghome)))
+					// nolint:lll,gosec
+					AddReportEntry(fmt.Sprintf("Detect XDG_CONFIG_HOME: %s, generated plugin path: %s", xdghome, plgPath), ReportEntryVisibilityAlways)
+				} else {
+					Expect(plgPath).To(Equal(fmt.Sprintf("%s/Library/Application Support/kubebuilder/plugins", homePath)))
+				}
```

Comment:
IMO these should be tests that are separated even further. Ideally we don't have to do this `if ... else` check and the xdghome check is in a separate test case. 

Something we could do to verify this is working appropriately for when $XDGHOME isn't set is to get the original value, unset it before each test, reset the original value after each test. Luckily Ginkgo v2 has a handy `GinkgoT().Setenv(...)` function we can use to do this for us.